### PR TITLE
Gradle tests Liberty, Payara, Helidon

### PR DESCRIPTION
**Pull request by Karm**
Gradle:
Removes Tomee and Wildfly and Quarkus until properly implemented in Starter.
Adds tests for Helidon, Payara and Liberty.

### CODE REVIEW

**Comment by Karm** [comment]
All Liberty tests are failing for me now due to ports already in use...

```
[ERROR   ] CWWKO0221E: TCP Channel defaultHttpEndpoint initialization did not succeed.  The socket bind did not succeed for host localhost and port 9081.  The port might already be in use.  Exception Message:

[ERROR   ] CWWKO0221E: TCP Channel defaultHttpEndpoint-ssl initialization did not succeed.  The socket bind did not succeed for host localhost and port 9444.  The port might already be in use.  Exception

[ERROR   ] CWWKO0221E: TCP Channel defaultHttpEndpoint initialization did not succeed.  The socket bind did not succeed for host localhost and port 9080.  The port might already be in use.  Exception Message: Address already in use

[ERROR   ] CWWKO0221E: TCP Channel defaultHttpEndpoint-ssl initialization did not succeed.  The socket bind did not succeed for host localhost and port 9443.  The port might already be in use.  Exception Message: Address already in use
```

Not sure what brought that in. Apparently port shift for service-b does not work?

**Comment by Karm** [comment]
The problem is Liberty + Gradle.  Ctrl+C simply does not terminate it. Servers keep running in the background, occupying ports :-(

**Comment by Karm** [comment]
+ JWT on Liberty fails when built and ran with Gradle:

```
[AUDIT   ] CWWKF0011I: The defaultServer server is ready to run a smarter planet. The defaultServer server started in 7.795 seconds.
[ERROR   ] CWWKS5524E: The MicroProfile JWT feature encountered an error while creating a JWT by using the [jwtUserConsumer] configuration and the token included in the request. CWWKS6031E: The JSON Web Token (JWT) consumer [jwtUserConsumer] cannot process the token string. CWWKS6022E: The issuer [https://server.example.com] of the provided JSON web token (JWT) is not listed as a trusted issuer in the [jwtUserConsumer] JWT configuration. The trusted issuers are [${jwt.issuer}]. 
```

Definitley `${jwt.issuer}` looks suspicious...

**Comment by Karm** [comment]
@Emily-Jiang  I tried

```
ext  {
    liberty.server.var.'default.http.port' = '9081'
    liberty.server.var.'project.name' = projectName
    liberty.server.var.'app.context.root' = '/'
    liberty.server.var.'jwt.issuer' = 'https://server.example.com'
}
```

and it didn't help to propagate it to  https://github.com/eclipse/microprofile-starter/blob/master/src/main/resources/files/liberty/server.xml.tpl#L28 and my Liberty x Gradle support kinda ends here :-)  I would like you to take a look and help, if you may.

**Comment by Emily-Jiang** [comment]
@Karm for the ctl+c issue, can you issue `gradlew libertyStop` to stop the server?

**Comment by Emily-Jiang** [comment]
> @Emily-Jiang I tried
> 
> ```
> ext  {
>     liberty.server.var.'default.http.port' = '9081'
>     liberty.server.var.'project.name' = projectName
>     liberty.server.var.'app.context.root' = '/'
>     liberty.server.var.'jwt.issuer' = 'https://server.example.com'
> }
> ```
> 
> and it didn't help to propagate it to https://github.com/eclipse/microprofile-starter/blob/master/src/main/resources/files/liberty/server.xml.tpl#L28 and my Liberty x Gradle support kinda ends here :-) I would like you to take a look and help, if you may.

I tried and it does work.

**Comment by Emily-Jiang** [comment]
https://github.com/eclipse/microprofile-starter/pull/429 is to fix service-b

**Comment by Karm** [comment]
@Emily-Jiang  Any idea how to stop Liberty so as  org.gradle.launcher.daemon.bootstrap.GradleDaemon process stops too?

The test suite has to forcibly kill those processes now.  

How do you do it locally?  (it's reproducible locally without TS too)

**Comment by Emily-Jiang** [comment]
Do you mean that you could not do `gradlew libertyStop`?

**Comment by Karm** [comment]
It looks ./gradlew libertyStop did it.

Let the tests run...

File: src/test/resources/readme_examples/he-service-a-gradle-readme.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
+# MicroProfile generated Application
+
+## Introduction
+
+MicroProfile Starter has generated this MicroProfile application for you.
+
+The generation of the executable jar file can be performed by issuing the following command
+
+
+
```
Comment (Emily-Jiang) [inline src/test/resources/readme_examples/he-service-a-gradle-readme.md:10]: Why do we need so many blank lines for the command? The same command applies to all the other command as well.

Comment (Karm) [reply src/test/resources/readme_examples/he-service-a-gradle-readme.md:10]: Templating emgine leaves them there. I reduced their number in this pr.

The file you are commenting on is a static test for the 'parser'. Ignore it.

**Comment by Emily-Jiang** [comment]
> @Emily-Jiang Any idea how to stop Liberty so as org.gradle.launcher.daemon.bootstrap.GradleDaemon process stops too?
> 
> The test suite has to forcibly kill those processes now.
> 
> How do you do it locally? (it's reproducible locally without TS too)

I talked with the Open Liberty team who owns the gradle plugin and confirmed this is a known issue. The current workaround is to use `gradlew libertyStop` or killing the process manually.

**Review by Emily-Jiang — APPROVED** [review]
LGTM! Thanks @Karm

**Pull request merged**